app/controllers/concerns/course_basic_detail_concern.rb

```diff
@@ -89,18 +104,73 @@ def build_previous_course_creation_params
   end
 
   def next_step(current_step:)
-    if current_step == :outcome
+    next_step = NextCourseCreationStepService.new.execute(current_step: current_step)
+    next_page = course_creation_path_for(next_step)
+
+    if next_page.nil?
+      raise "No path defined for next step: #{next_step}"
+    end
+
+    next_page
+  end
+
+  def course_creation_path_for(page)
+    case page
+    when :apprenticeship
+      new_provider_recruitment_cycle_courses_apprenticeship_path(
+        params[:provider_code],
+        params[:recruitment_cycle_year],
+        course: course_params,
+      )
+      # Currently the page isnt built - so skip
+    when :location
       new_provider_recruitment_cycle_courses_entry_requirements_path(
         params[:provider_code],
         params[:recruitment_cycle_year],
         course: course_params,
       )
-    elsif current_step == :entry_requirements
+    when :entry_requirements
+      new_provider_recruitment_cycle_courses_entry_requirements_path(
+        params[:provider_code],
+        params[:recruitment_cycle_year],
+        course: course_params,
+      )
+    when :outcome
       new_provider_recruitment_cycle_courses_outcome_path(
         params[:provider_code],
         params[:recruitment_cycle_year],
         course: course_params,
       )
+    when :full_or_part_time
+      new_provider_recruitment_cycle_courses_study_mode_path(
+        params[:provider_code],
+        params[:recruitment_cycle_year],
+        course: course_params,
+      )
+    when :applications_open
+      new_provider_recruitment_cycle_courses_applications_open_path(
+        params[:provider_code],
+        params[:recruitment_cycle_year],
+        course: course_params,
+      )
+    when :start_date
+      new_provider_recruitment_cycle_courses_start_date_path(
+        params[:provider_code],
+        params[:recruitment_cycle_year],
+        course: course_params,
+      )
+    when :age_range
+      new_provider_recruitment_cycle_courses_age_range_path(
+        params[:provider_code],
+        params[:recruitment_cycle_year],
+        course: course_params,
+      )
+    when :confirmation
+      confirmation_provider_recruitment_cycle_courses_path(
+        params[:provider_code],
+        params[:recruitment_cycle_year],
+        course: course_params,
+      )
```

Comment:
I was just looking into Code Climate's complaint about this method being too long. It turns out we can cut this down by quite a bit, actually:

```
  def course_creation_path_for(page)
    path_params = params.slice(:provider_code, :recruitment_cycle_year)
                    .merge(course: course_params)
    case page
    when :apprenticeship
      new_provider_recruitment_cycle_courses_apprenticeship_path(path_params)
    when :location
      new_provider_recruitment_cycle_courses_entry_requirements_path(path_params)
    ...
```

This has the benefit in that it's easier to reason about the params a little bit ... before I found myself wondering if they were all the same, etc.

Also, it seems we don't even need `provider_code` and `recruitment_cycle_year`, from my testing, so I _think_ this would be fine too:

```
    path_params = { course: course_params }
```

Just some thoughts. I'm happy to worry about this later if you just want to get it merged.

Reply:
Huh - I thought I tested it without the `provider_code` and `recruitment_cycle_year` but apparently I didn't. That works well, i'll make the changes and commit :tada:

Reply:
:tada: All green! :shipit:  🚀